dom/events/scrolling/scroll_support.js

```diff
@@ -83,3 +83,18 @@ function touchScrollInTarget(pixels_to_scroll, target, direction, pause_time_in_
 function touchFlingInTarget(pixels_to_scroll, target, direction) {
   touchScrollInTarget(pixels_to_scroll, target, direction, 0 /* pause_time */);
 }
+
+async function mouseActionsInTarget(target, origin, delta, pause_time_in_ms = 100) {
+  return new test_driver.Actions()
+    .addPointer("pointer1", "mouse")
+    .pointerMove(origin.x, origin.y, { origin: target })
+    .pointerDown()
+    .pointerMove(origin.x + delta.x, origin.y + delta.y, { origin: target })
+    .pointerMove(origin.x + delta.x * 2, origin.y + delta.y * 2, { origin: target })
+    .send()
+    .then(() => {
+      setTimeout(() => {
+        return new test_driver.Actions().pointerUp().send();
```

Comment:
Instead of this why not just use .pause function on the Actions object? Like
`.pointermove(...).pause(pause_time_in_ms).pointerUp().send();`

But that aside why did you need this delay in the first place?

Reply:
Sorry, I was working on WebKit. It seems the test_driver interfaces in WebKit does not update. 
In the old interfaces, `dispatchMouseActions()` in resources/testdriver-vendor.js omits `pause`.
```
function dispatchMouseActions(actions)
{
    if (!window.eventSender)
        return Promise.reject(new Error("window.eventSender is undefined."));

    return new Promise(resolve => {
        setTimeout(() => {
            eventSender.dragMode = false;

            for (let action of actions) {
                switch (action.type) {
                case "pointerMove":
                    let origin = { x: 0, y: 0 };
                    if (action.origin instanceof Element) {
                        const bounds = action.origin.getBoundingClientRect();
                        logDebug(() => `${action.origin.id} [${bounds.left}, ${bounds.top}, ${bounds.width}, ${bounds.height}]`);
                        origin.x = bounds.left + 1;
                        origin.y = bounds.top + 1;
                    }
                    logDebug(() => `eventSender.mouseMoveTo(${action.x + origin.x}, ${action.y + origin.y})`);
                    eventSender.mouseMoveTo(action.x + origin.x, action.y + origin.y);
                    break;
                case "pointerDown":
                    logDebug(() => `eventSender.mouseDown()`);
                    eventSender.mouseDown(action.button);
                    break;
                case "pointerUp":
                    logDebug(() => `eventSender.mouseUp()`);
                    eventSender.mouseUp(action.button);
                    break;
                case "pause":
                    // FIXME: What should we do here?
                    break;
                default:
                    return Promise.reject(new Error(`Unknown action type "${action.type}".`));
                }
            }
            resolve();
        });
    });
}
```
Without `pause`, we can't generate effective drag actions, so I used `setTimeout`here.

And I tried to confirm `pause` is properly handled in the latest code, but I couldn't find it. Sorry...
But I agree to change it to `pause(pause_time_in_ms).pointerUp().send();` here.